source/navitiacommon/navitiacommon/models.py

```diff
@@ -401,6 +401,19 @@ def get_from_id_or_name(cls, id=None, name=None):
         else:
             raise Exception({'error': 'instance is required'}, 400)
 
+    def delete_dataset(self):
+        jobs = db.session.query(Job).filter(Job.instance_id == self.id).all()
+        for j in jobs:
+            for dataset in j.data_sets.all():
```

Comment:
this isn't very efficient, you will get every job then iterate over all the datasets of all these job.
You could do a query to only get the dataset of type POI, then delete

Reply:
the new API specifies one instance in the URL. By querying all POI datasets, those from regions not specified in the query will be deleted too

Reply:
and you can do `AND` in the query

Reply:
Changed